lib/NetPacksLib.cpp

```diff
@@ -594,7 +594,11 @@ DLL_LINKAGE void HeroRecruited::applyGs( CGameState *gs )
 
 	h->setOwner(player);
 	h->pos = tile;
-	h->movement =  h->maxMovePoints(true);
+	bool fresh = !h->isInitialized();

```

Comment:
Weird non-logical naming:
1) fresh = not initialized
2) initialized if mana and MP are _not_ initialized